django/core/signing.py

```diff
@@ -269,12 +277,26 @@ def unsign(self, value, max_age=None):
         """
         result = super().unsign(value)
         value, timestamp = result.rsplit(self.sep, 1)
-        timestamp = b62_decode(timestamp)
-        if max_age is not None:
-            if isinstance(max_age, datetime.timedelta):
-                max_age = max_age.total_seconds()
-            # Check timestamp is not older than max_age
-            age = time.time() - timestamp
-            if age > max_age:
-                raise SignatureExpired("Signature age %s > %s seconds" % (age, max_age))
+        self.payload_timestamp = b62_decode(timestamp)
+        self._verify_max_age(max_age)
         return value
+
+    def unsign_object(self, signed_obj, max_age=None, **kwargs):
+        if self.expiration_key is None:
+            return super().unsign_object(signed_obj, max_age=max_age, **kwargs)
+
+        # If using expiration key use it's value when present
+        value = super().unsign_object(signed_obj, **kwargs, max_age=None)
```

Comment:
I do not think that we should skip `max_age`. I have the feeling that `max_age` should be verified no matter what and the `expiration_key` should be an additional validation -- if one wouldn't like to have `max_age` validation then the caller wouldn't have to pass it in.